### DISCUSSION TITLE
Does detect.py grayscale or B/W?

### DISCUSSION THREAD

**Discussion by danielsoy** [ultralytics/yolov5#5411] [Q&A] [answered]
Hello Glenn!

Does detect.py grayscale or B/W preprocessing before pass the data from the webcam, to the model for inference?

In case it does not, is it possible to add it?

Which is the variable name, and at what point should I pick it up to add this preprocessing stage if possible?

Thanks!

**Comment by glenn-jocher** [comment, marked as answer]
@danielsoy detect.py YOLOv5 dataloaders use cv2.imread() which will automatically convert 1ch greyscale to 3ch greyscale prior to inference.

**Reply by danielsoy** [reply]
I guess  i found it inside datasets.py

Thx again!
